src/NuGet.Core/NuGet.Commands/RestoreCommand/DownloadDependencyResolutionResult.cs

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using NuGet.DependencyResolver;
+using NuGet.Frameworks;
+using NuGet.LibraryModel;
+
+namespace NuGet.Commands
+{
+    public class DownloadDependencyResolutionResult
+    {
+        /// <summary>
+        /// The framework for which the resolution happened.
+        /// </summary>
+        public NuGetFramework Framework { get; }
+
+        /// <summary>
+        /// A list of library ranges and the dependencies.
+        /// </summary>
+        public IList<Tuple<LibraryRange, RemoteMatch>> Dependencies { get; }
+
+        /// <summary>
+        /// The set of matches that are being installed.
+        /// </summary>
+        public ISet<RemoteMatch> Install { get; }
+
+        /// <summary>
+        /// A set of unresolved library ranges.
+        /// </summary>
+        public ISet<LibraryRange> Unresolved { get; }
```

Comment:
similar to above, this could also be just string to indicate unresolved package ids.

Reply:
We need the package id and version for the error/warning experience later on. The library range model is pretty minimal one, containing the Id, version + type. The type is unnecessary in this case, but I decided against creating a new type and propagating it through a whole new set of APIs.